cli/azd/pkg/apphost/manifest.go

```diff
@@ -36,6 +36,9 @@ type Resource struct {
 	// BuildArgs is present on a dockerfile.v0 resource and is the --build-arg for building the docker image.
 	BuildArgs map[string]string `json:"buildArgs,omitempty"`
 
+	// Args is optionally present on project.v0 and dockerfile.v0 resources and are the arguments to pass to the container.
+	Args *[]string `json:"args,omitempty"`
```

Comment:
Why do we use a pointer to slice here?  
Makes me wonder if this is some sort of redundant.  I might be wrong, but, isn't the `omitempy` rule from json let a slice as `nil` if the `args` node is not found in the json?
isn't a slice capable enough to cover the 3 states: nil (not in the json), empty (in json but empy array) and withValues (in json with values)?  Are we trying to cover scenarios like `args: null` in json?

Reply:
I changed this such that it is now `[]string`, I'm not sure why I did this (I think it was because I normally just use *T if T is optional, but I see now we don't do that for maps or other arrays). Thanks for pointing this out, @vhvb1989.